core/test/log/logger.cpp

```diff
@@ -83,6 +83,21 @@ TEST(DummyLogged, CanAddMultipleLoggers)
 }
 
 
+TEST(DummyLogged, CanRemoveLogger)
+{
+    auto exec = gko::ReferenceExecutor::create();
+    DummyLoggedClass c;
+    auto r = gko::share(
+        gko::log::Record::create(exec, gko::log::Logger::all_events_mask));
+    c.add_logger(r);
+    c.add_logger(gko::log::Stream<>::create(
+        exec, gko::log::Logger::all_events_mask, std::cout));
+
+    c.remove_logger(gko::lend(r));
+
+    ASSERT_EQ(c.get_num_loggers(), 1);
```

Comment:
return value 1 because this is the default logger?

Reply:
We attach 2 loggers, we remove 1, the number of loggers left is 1.